chrome-extension/content.js

```diff
@@ -140,12 +159,73 @@ async function createSidebar(content) {
     const buttonContainer = customCreateElement('div', ['has-text-centered', 'mt-1']);
     const addButton = customCreateElement('button', ['button', 'is-link'], 'Add Clipping');
     addButton.type = 'submit';
+    addButton.id = 'lobster-submit-button';
     buttonContainer.appendChild(addButton);
     form.appendChild(buttonContainer);
     form.appendChild(userEmailNote);
   });
 }
 
+/**
+ * Helper to create a small new category form
+ * @param {String} userId 
+ */
+function createNewCategoryForm(userId) {
+  const newCategoryForm = document.createElement('form');
+  newCategoryForm.addEventListener('submit', (event) => {
+    event.preventDefault();
+    addNewCategory(userId);
+  });
+
+  const newCategoryBar = customCreateElement('div', ['field', 'has-addons']);
+  const inputControl = customCreateElement('div', ['control']);
+  const categoryInput = customCreateElement('input', ['input']);
+  categoryInput.setAttribute('placeholder', 'Add new category');
+  categoryInput.setAttribute('type', 'text');
+  categoryInput.id = 'newCategoryInput';
+  inputControl.appendChild(categoryInput);
+
+  const buttonControl = customCreateElement('div', ['control']);
+  const categoryButton = customCreateElement('button', ['button'], 'Create');
+  categoryButton.id = 'newCategoryButton';
+  categoryButton.type = 'submit';
+  buttonControl.appendChild(categoryButton);
+  newCategoryBar.append(inputControl, buttonControl);
+  newCategoryForm.appendChild(newCategoryBar);
+
+  return newCategoryForm;
+}
+
+/**
+ * Helper to add a new category to db upon form submission
+ * and add a new option to categoryDropdown
+ * @param {String} userId 
+ */
+async function addNewCategory(userId) {
+  const newCategoryTitle = document.getElementById('newCategoryInput').value;
+  const newCategory = {userId: userId,
+                       title: newCategoryTitle};
+  try {
+    const response = await fetch(serverUrl + '/category/createCategory', {
+      method: 'POST',
+      body: JSON.stringify(newCategory),
+      headers: { 'Content-type': 'application/json' },
+    });
+
+    if (response.status !== 200) {
+      throw new Error(response.statusMessage);
+    } else {
+      const newOption = customCreateElement('option', [], newCategoryTitle);
+      const newCategoryId = await response.json();
+      newOption.value = newCategoryId;
+      categoryDropdown.options.add(newOption);
+      document.getElementById('newCategoryInput').value = '';
+    }
```

Comment:
You don't need the else block since the throw would have forced it to exit this code block